lib/state/preferences/reducer.ts

```diff
@@ -0,0 +1,30 @@
+import * as A from '../action-types';
+
+type Preferences = {
+  analytics_enabled: boolean | null;
+};
+
+const uninitializedPreferences = {
+  analytics_enabled: null,
+};
+
+const preferences: A.Reducer<Preferences> = (
+  state = uninitializedPreferences,
+  action
+) => {
+  switch (action.type) {
+    case 'TOGGLE_ANALYTICS':
+      return {
+        ...state,
+        analytics_enabled:
+          typeof action.sendAnalytics !== 'undefined' ||
+          !state.analytics_enabled,
+      };
```

Comment:
I'm confused on what we're trying to accomplish here. It reads "if the action specifies whether or not to turn on the analytics then turn them on regardless, otherwise flip what we previously had"

should it be this?

```ts
typeof action.sendAnalytics !== 'undefined' ? action.sendAnalytics || !state.analytics_enabled
```

or this?

```ts
action.hasOwnProperty( 'sendAnalytics' ) ? action.sendAnalytics : !state.analytics_enabled
```